tests/conftest.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import shutil
+import subprocess
+from pathlib import Path, PurePath
+from typing import Optional, Protocol
```

Comment:
Protocol is only in Python 3.8. I believe you can change `basepython` in `tox.ini` to `python3.8` to fix the CI failures. Otherwise, you'd need to use `typing_extensions`.

Reply:
Right. Went with forcing 3.8.